ReactWindows/ReactNative/Storage/FutureAccessList.cs

```diff
@@ -0,0 +1,146 @@
+
+        /// <summary>
+        /// Checks the syntax of the FAL URI. It doesn't check if
+        /// this URI corresponds to an entry in the FAL.
+        /// </summary>
+        public static bool IsValidURI(string uri)
+        {
+            return uri.StartsWith(Scheme);
+        }
+
+        /// <summary>
+        /// Stores a file or folder object in the future access list cache.
+        /// The capacity of the cache is limited (up to about 1000 items) and
+        /// if necessary old items can be removed to free space for new ones.
+        /// </summary>
+        public static string Add(IStorageItem item)
+        {
+            var token = GetAccessToken(item);
+            return Scheme + Uri.EscapeUriString(token);
+        }
+
+        /// <summary>
+        /// Removes the file or folder associated with this URI.
+        /// </summary>
+        public static void Remove(string uri)
+        {
+            if (TryExtractAccessToken(uri, out var token))
+            {
+                StorageApplicationPermissions.FutureAccessList.Remove(token);
+            }
+        }
+
+        /// <summary>
+        /// Resolves a URI returned by the Add(file) method to the file object.
+        /// The file object remains in the cache and can be retrieved again later.
+        /// </summary>
+        public static async Task<StorageFile> GetFileAsync(string uri)
+        {
+            var token = ExtractAccessToken(uri);
+            return await StorageApplicationPermissions.FutureAccessList.GetFileAsync(token);
+        }
+
+        /// <summary>
+        /// Resolves a URI returned by the Add(folder) method to the folder object.
+        /// The folder object remains in the cache and can be retrieved again later.
+        /// </summary>
+        public static async Task<StorageFolder> GetFolderAsync(string uri)
+        {
+            var token = ExtractAccessToken(uri);
+            return await StorageApplicationPermissions.FutureAccessList.GetFolderAsync(token);
+        }
+
+        /// <summary>
+        /// Checks if FAL contains a file or folder with the given URI.
+        /// Returns false If the URI is invalid.
+        /// </summary>
+        public static bool ContainsItem(string uri)
+        {
+            if (TryExtractAccessToken(uri, out var token))
+            {
+                return StorageApplicationPermissions.FutureAccessList.ContainsItem(token);
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private static bool TryExtractAccessToken(string uri, out string token)
+        {
+            if (!uri.StartsWith(Scheme))
+            {
+                token = null;
+                return false;
+            }
+            else
+            {
+                token = Uri.UnescapeDataString(uri.Substring(Scheme.Length));
+                return true;
+            }
+        }
+
+        private static string ExtractAccessToken(string uri)
```

Comment:
>ExtractAccessToken [](start = 30, length = 18)

Pass a parameter here, `paramName`, so that you can pass that to the `ArgumentException`.

Reply:
Fixed